src/math/log_e.c

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2021 Intel Corporation. All rights reserved.
+//
+// Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+//
+//
+
+#include <sof/math/log.h>
+
+/**
+ * Base-e logarithm  loge(x)
+ *
+ * loge = (u) computes the base-e logarithm of
+ * u using lookup table.
+ * input u must be scalar/real number and positive
+ *
+ * +------------------+-----------------+--------+--------+
+ * | inpfxp	      |loge (returntype)| inpfxp |  loge  |
+ * +----+-----+-------+----+----+-------+--------+--------+
+ * |WLen| FLen|Signbit|WLen|FLen|Signbit| Qformat| Qformat|
+ * +----+-----+-------+----+----+-------+--------+--------+
+ * | 32 | 0   |	 0    | 32 | 27 |   0	| 32.0	 | 5.27	  |
+ * +------------------+-----------------+--------+--------+
+ *
+ * Arguments	: uint32_t numerator [1 to 4294967295- Q32.0]
+ * Return Type	: uint32_t UQ5.27 [ 0 to 22.1808076352]
+ */
+uint32_t log_e_int32(uint32_t numerator)
```

Comment:
In C library the function names are log() and log10(), would just log_int32() and log10_int32() be more familiar looking functions names?

Reply:
Done.